web/Controllers/PropertyController.cs

```diff
@@ -140,6 +140,24 @@ public ActionResult AddNewProperty_Cancel()
         // Go to View Properties page
         return RedirectToAction("ViewProperties", "Landlord");
     }
+    
+    [Authorize(Roles = "Landlord")]
+    public ActionResult DeleteProperty(int propertyId)

```

Comment:
This should be a POST - having GET requests cause side effects can cause issues. It will normally be OK in practice, but the browser is within its rights to prefetch the link, retry the request or cache the response.

You'll need to change the `<input>` to be in a `<form method="POST">` tag in the view